tests/notebooks/training/training-integration.ipynb

```diff
@@ -1,897 +1,918 @@
+     },
+     {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+       "scrolled": true
+      },
+      "outputs": [],
+      "source": [
+       "print(\"Job:\", paddlejob.metadata.name, end=\"\\n\\n\")\n",
+       "print(\"Job Spec:\", paddlejob.spec, sep=\"\\n\", end=\"\\n\\n\")\n",
+       "print(\"Job Status:\", paddlejob.status, sep=\"\\n\", end=\"\\n\\n\")"
+      ]
+     },
+     {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+       "### Get PaddleJob Training logs\n",
+       "Get and print the training logs of the PaddleJob with the training steps "
+      ]
+     },
+     {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+       "# set is_master to False because this example does not include a master replica type\n",
+       "print_training_logs(client, PADDLEJOB_NAME, container=PADDLEJOB_CONTAINER, is_master=False)"
+      ]
+     },
+     {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+       "### Delete PaddleJob\n",
+       "\n",
+       "Delete the created PaddleJob."
+      ]
+     },
+     {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+       "client.delete_paddlejob(name=PADDLEJOB_NAME)"
+      ]
+     },
+     {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+       "@retry(\n",
+       "    wait=wait_exponential(multiplier=2, min=1, max=10),\n",
+       "    stop=stop_after_attempt(30),\n",
+       "    reraise=True,\n",
+       ")\n",
+       "def assert_paddlejob_removed(client, job_name):\n",
+       "    \"\"\"Wait for PaddleJob to be removed.\"\"\"\n",
+       "    # fetch the existing PaddleJob names\n",
+       "    # verify that the Job was deleted successfully\n",
+       "    jobs = {job.metadata.name for job in client.list_paddlejobs()}\n",
+       "    assert job_name not in jobs, f\"Failed to delete PaddleJob {job_name}!\""
+      ]
+     },
+     {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+       "tags": [
+        "raises-exception"
+       ]
+      },
+      "outputs": [],
+      "source": [
+       "# wait for PaddleJob to be removed successfully\n",
+       "assert_paddlejob_removed(client, PADDLEJOB_NAME)"
+      ]
+     }
+    ],
+    "metadata": {
+     "kernelspec": {
+      "display_name": "Python 3 (ipykernel)",
+      "language": "python",
+      "name": "python3"
+     },
+     "language_info": {
+      "codemirror_mode": {
+       "name": "ipython",
+       "version": 3
+      },
+      "file_extension": ".py",
+      "mimetype": "text/x-python",
+      "name": "python",
+      "nbconvert_exporter": "python",
+      "pygments_lexer": "ipython3",
+      "version": "3.11.9"
+     }
+    },
+    "nbformat": 4,
+    "nbformat_minor": 4
+   }
```

Comment:
the pytorchjob and paddlejob parts we can refactor now, these ones are passing with the changes
the issue is open for tfjob only
@orfeas-k